src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,332 @@
+                }
+            }
+
+            this.#dynamicClients = newDynamicClients;
+            this.#lastFallbackClientRefreshTime = Date.now();
+        } catch (err) {
+            console.warn(`Fail to build fallback clients, ${err.message}`);
+        }
+    }
+
+    #newConfigurationClient(endpoint) {
+        if (this.#credential) {
+            return new AppConfigurationClient(endpoint, this.#credential, this.#clientOptions);
+        }
+
+        const connectionStr = buildConnectionString(endpoint, this.#secret, this.#id);
+        return new AppConfigurationClient(connectionStr, this.#clientOptions);
+    }
+
+    #isFallbackClientDiscoveryDue(dateTime) {
+        return dateTime >= this.#lastFallbackClientRefreshAttempt + MinimalClientRefreshInterval
+            && (!this.#dynamicClients
+                || this.#dynamicClients.every(client => dateTime < client.backoffEndTime)
+                || dateTime >= this.#lastFallbackClientRefreshTime + FallbackClientRefreshExpireInterval);
+    }
+}
+
+/**
+ * Query SRV records and return target hosts.
+ * @param {string} host - The host to query.
+ * @returns {Promise<string[]>} - A promise that resolves to an array of target hosts.
+ */
+async function querySrvTargetHost(host) {
+    const results: string[] = [];
+    let dns;
+
+    if (isFailoverableEnv()) {
+        dns = require("dns/promises");
+    } else {
+        return results;
+    }
+
+    try {
+        // Look up SRV records for the origin host
+        const originRecords = await dns.resolveSrv(`${TCP_ORIGIN}.${host}`);
+        if (originRecords.length === 0) {
+            return results;
+        }
+
+        // Add the first origin record to results
+        const originHost = originRecords[0].name;
+        results.push(originHost);
+
+        // Look up SRV records for alternate hosts
+        let index = 0;
+        let moreAltRecordsExist = true;
+        while (moreAltRecordsExist) {
+            const currentAlt = `${ALT}${index}`;
+            try {
+                const altRecords = await dns.resolveSrv(`${currentAlt}.${TCP}.${originHost}`);
+                if (altRecords.length === 0) {
+                    moreAltRecordsExist = false;
+                    break; // No more alternate records, exit loop
+                }
+
+                altRecords.forEach(record => {
+                    const altHost = record.name;
+                    if (altHost) {
+                        results.push(altHost);
+                    }
+                });
+                index++;
+            } catch (err) {
+                if (err.code === "ENOTFOUND") {
+                    break; // No more alternate records, exit loop
+                } else {
+                    throw new Error(`Failed to lookup alternate SRV records: ${err.message}`);
+                }
+            }
+        }
+    } catch (err) {
+        throw new Error(`Failed to lookup origin SRV records: ${err.message}`);
```

Comment:
If line213 fail, will the error be wrapped into "Failed to lookup origin SRV records: failed to lookup alternate SRV records:  xxxx"?

Reply:
If error occurs when query originRecords, it will not execute the altRecords query.

Reply:
I don't get you, you throw error in line213, and it might be watched by line217,

Reply:
Ohhhhh yes, if failed inside, the error will be propagated. The error msg outside should be update.